### PR TITLE
[PM-10589] Update database defaults for collection management settings

### PR DESCRIPTION
**Pull request by r-tome**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-10589

## 📔 Objective

> When we added LimitCollectionCreationDeletion and AllowAdminAccessToAllCollectionItems, we set them to default to 1, being the transitional setting.
> 
> That no longer applies because all orgs are being created with these set to 0 by the code. We should probably update this at the db level to reflect that, in particular because AllowAdminAccessToAllCollectionItems = 1 is the more permissive setting which is not intended.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by eliykat — CHANGES_REQUESTED** [review]
Thanks for doing this! There is also a default constraint on the column: https://github.com/bitwarden/server/blob/main/src/Sql/dbo/Tables/Organization.sql#L55

**Comment by r-tome** [comment]
> Thanks for doing this! There are also default constraints on the columns: https://github.com/bitwarden/server/blob/main/src/Sql/dbo/Tables/Organization.sql#L54-L55

I forgot to check those, thank you!

File: util/Migrator/DbScripts/2024-08-13_00_UpdateCollectionManagementSettingsDbDefaults.sql [hunk 1]
```diff
@@ -0,0 +1,312 @@
+-- Objective: 'LimitCollectionCreationDeletion' and 'AllowAdminAccessToAllCollectionItems' columns were set to 1 by
+-- default as a transitional measure. This script updates the default values to 0 to mirror the values set in code.
+
+-- Step 1: Drop existing default constraints
+ALTER TABLE [dbo].[Organization] DROP CONSTRAINT [DF_Organization_LimitCollectionCreationDeletion];
+GO
+ALTER TABLE [dbo].[Organization] DROP CONSTRAINT [DF_Organization_AllowAdminAccessToAllCollectionItems];
+GO
```
Comment (eliykat) [inline util/Migrator/DbScripts/2024-08-13_00_UpdateCollectionManagementSettingsDbDefaults.sql:8]: We should check if the constraint exists before trying to drop it: [example from our code](https://github.com/bitwarden/server/blob/main/util/Migrator/DbScripts/2024-06-11_00_FixProviderInvoiceItem.sql#L9-L15), [related SO answer](https://stackoverflow.com/a/13186918). Although that doesn't discriminate between tables.

@rkac-bw what's your recommendation?

File: util/Migrator/DbScripts/2024-08-13_00_UpdateCollectionManagementSettingsDbDefaults.sql [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+
+-- Step 2: Add new default constraints with default value of 0
+ALTER TABLE [dbo].[Organization] ADD CONSTRAINT [DF_Organization_LimitCollectionCreationDeletion] DEFAULT (0) FOR [LimitCollectionCreationDeletion];
+GO
+ALTER TABLE [dbo].[Organization] ADD CONSTRAINT [DF_Organization_AllowAdminAccessToAllCollectionItems] DEFAULT (0) FOR [AllowAdminAccessToAllCollectionItems];
+GO
```
Comment (eliykat) [inline util/Migrator/DbScripts/2024-08-13_00_UpdateCollectionManagementSettingsDbDefaults.sql:14]: Similarly, need to check that the column exists before adding the constraint.

**Review by eliykat — CHANGES_REQUESTED** [review]
Mostly lgtm, just a few questions around our usual practice of checking that things exist before changing them.

**Review by eliykat — APPROVED** [review]
Looks good to me, thanks!

File: util/Migrator/DbScripts/2024-08-13_00_UpdateCollectionManagementSettingsDbDefaults.sql (hunk 1 above)
Comment (rkac-bw) [reply util/Migrator/DbScripts/2024-08-13_00_UpdateCollectionManagementSettingsDbDefaults.sql]: It's much cleaner to check if exists

**Review by rkac-bw — APPROVED** [review]
lgtm

**Pull request merged**